veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,656 @@
+will use the architecture specified in the `spec.template.spec.source.registry.platform.architecture` field of the
+`DataImportCron` CR, if provided.
+
+The importer will pull the image corresponding to the specified architecture and loads it into the `VolumeSnapshot`.
+1. if the `pullMethod` is `pod`, then the cdi-importer will use the platform field to pull the architecture specific
+   image.
+   > **Note**: when the cdi-importer is trying to pull an architecture specific image in a registry, it will set the
+   > `os` field of the image platform to `linux`. The platform reflecting the runtime environment rather than the image
+   > contents (i.e., VM images in KubeVirt may represent different operating systems).
+
+2. when the `pullMethod` is `node`, the cdi-controller creates a pod with an init container with the required image
+   letting the node's CRI to perform the actual pull. In this case the CRI will pull the image matches to its own 
+   architecture. the cdi-controller will make sure that the pod will run on a node with the required architecture by
+   modifying the pod's `nodeSelector` or `nodeAffinity` fields, to force scheduling on a node with the
+   `kubernetes.io/arch` label, matching the architecture specified in the
+   `spec.template.spec.source.registry.platform.architecture` field.
+
+   if there is no available node with the required architecture, the cdi-controller will report this failure in a
+   condition on the corresponding `DataImportCron` CR.
+
+If the image is not a manifest or does not include the specified architecture, the CDI will report this failure in a
+condition on the corresponding `DataImportCron` CR.
+
+If the `spec.template.spec.source.registry.platform.architecture` field is not populated, the cdi-importer will pull the 
+image without specifying an architecture, maintaining the current behavior.
+
+### Cloning the Image into a Virtual Machine's PVC
+In order to support the new "pointer" `DataSource` CR, The cdi-cloner, when populating the virtual-machine PVC, will use
+the `status.source` field to find the right `VolumeSnapshot` or `PVC`, rather than the value of the `spec.source` field.
+
+### Creating a Virtual Machine Templates
+SSP will deploy all templates supported in the cluster, corresponding to the architectures listed in the new
+`spec.cluster.workloadArchitectures` `SSP` CR's field. SSP dynamically add or remove templates if the architecture list
+in the `SSP` CR changes.
+
+For each architecture-specific template, SSP will set the `DATA_SOURCE_NAME` to the name of the
+architecture-specific `DataSource`. The `DataSource` can be identified using its `template.kubevirt.io/architecture` and
+the `cdi.kubevirt.io/storage.import.datasource-name` labels.
+
+### Creating a Virtual Machine from a Template
+Assuming the user knows the architecture of the VM they want to create, the kubevirt-plugin (UI) will prompt the user to
```

Comment:
Do we need to make adjustments for VirtualMachinePreferences in the UI?

Reply:
Please suggest the require change.